### DISCUSSION TITLE
Adding forecast capabilities to the Ontario, Canada parser

### DISCUSSION THREAD

**Discussion by r24mille** [electricitymaps/electricitymaps-contrib#3601] [Ideas]
Hello! I've been looking at the Ontario parser a bit. Neither `fetch_consumption_forecast(...)`, `fetch_generation_forecast(...)`, `fetch_wind_solar_forecasts(...)`, nor `fetch_exchange_forecast(...)` have been implemented in the `CA_ON.py` parser. I believe the IESO has reasonable reports for these:

- `fetch_consumption_forecast(...)`
  - The [Predispatch Constrained Totals](http://reports.ieso.ca/public/PredispConstTotals/PUB_PredispConstTotals.xml).
- `fetch_generation_forecast(...)` and `fetch_wind_solar_forecasts(...)` 
  - The [Adequacy Report](http://reports.ieso.ca/public/Adequacy2/).
  - Although reports exist ~1 month into the future, only the current and next days contain the "Scheduled" element, which would be used as the data source (e.g. http://reports.ieso.ca/public/Adequacy2/PUB_Adequacy2_20211211_v72.xml).
  - The IESO [states that](http://reports.ieso.ca/docrefs/helpfile/Adequacy2_h2.pdf) the schedules for the next day are posted by approximately 11:15am. Anecdotally, I've seen "approximate" mean a little after 11:20am.
- `fetch_exchange_forecast(...)`
  - The [Adequacy Report](http://reports.ieso.ca/public/Adequacy2/) could also be used for imports/exports

I am willing and interested in developing this, if electricityMap would find value in it. Before sinking development time into it and opening pull requests, I wanted to confirm whether the maintainers of this repository would find value in those contributions. Thanks!

**Comment by pierresegonne** [comment]
Hey @r24mille ! sorry for the long delay!

For something like this, I'd recommend you open an issue with the label "parser buildable". Discussions are oriented more towards larger ideas and methodological discussions. 

If you have time and energy please go ahead and build these parsers :)
